test/unit/utils/UWebSocketTransport.spec.ts

```diff
@@ -10,270 +15,693 @@ describe('UWebSocketTransport', () => {
+    // Coverage for lines 300-301
+    it('should handle cleanup with pending timeouts', async () => {
+      const message = new Uint8Array([1])
+      mockSocket.send.mockReturnValue(UWebSocketSendResult.DROPPED)
+
+      const sendPromise = transport.sendMessage(message)
+      // Add catch to prevent unhandled rejection
+      ;(sendPromise as unknown as Promise<void>)?.catch(() => {})
+
+      // Let the first attempt process
+      await jest.advanceTimersByTimeAsync(0)
+
+      // Close before timeout triggers
       transport.close()
-      mockSocket.end.mockClear()
-      mockSocket.getUserData.mockReturnValue({ isConnected: false }) // Indicate connection is closed
-      transport.close() // Second close should not call end again
-      expect(mockSocket.end).not.toHaveBeenCalled()
+
+      // Wait for cleanup to complete
+      await jest.runAllTimersAsync()
+
+      await expect(sendPromise).rejects.toThrow('Connection closed')
+
+      // Ensure no more processing occurs
+      await jest.advanceTimersByTimeAsync(5000)
+      expect(mockSocket.send).toHaveBeenCalledTimes(1)
     })
   })
 
-  describe('Error Scenarios', () => {
-    it('should handle socket errors during queue processing', async () => {
+  describe('Transport Error Handling', () => {
+    it('should handle errors during message processing (send error)', async () => {
+      const message = new Uint8Array([1])
+      const networkError = new Error('Network error')
       mockSocket.send.mockImplementation(() => {
-        throw new Error('Socket error')
+        throw networkError
       })
 
-      const message = new Uint8Array([1])
-      await expect(transport.sendMessage(message)).rejects.toThrow('Failed to send message: Socket error')
-    })
+      const sendPromise = transport.sendMessage(message)
+
+      // Add catch after assertion to prevent Jest unhandled rejection error
+      ;(sendPromise as unknown as Promise<void>)?.catch(() => {})
+
+      // Process the send attempt that throws
+      await jest.advanceTimersByTimeAsync(0)
+
+      // Check rejection
+      await expect(sendPromise).rejects.toThrow('Failed to send message: Network error')
 
-    it('should handle queue timeout during processing', async () => {
-      jest.useFakeTimers()
-      mockSocket.send.mockReturnValue(0) // Force send failure
+      // Check logger call (Exact message and payload)
+      expect(mockLogs.getLogger('ws-transport').error).toHaveBeenCalledWith(
+        'Error sending message', // Exact message
+        {
+          transportId: expect.any(String),
+          error: 'Failed to send message: Network error'
+        }
+      )
+      // Check event emitter call
+      expect(errorListener).toHaveBeenCalledWith(new Error('Failed to send message: Network error'))
+    })
 
+    it('should handle cleanup during active message processing', async () => {
       const message = new Uint8Array([1])
+      mockSocket.send.mockReturnValue(UWebSocketSendResult.DROPPED)
```

Comment:
If I'm not wrong, isn't this testing the same as in [the lines 565](https://github.com/decentraland/social-service-ea/pull/91/files#diff-b567943d48312e33460faf772f889ffb94a09c02003ba44388e7ea94d9bd02a6R677)?